regions/src/iosMain/kotlin/com/privateinternetaccess/regions/internals/PersistenceRegionsDataSource.kt

```diff
@@ -1,26 +1,29 @@
 package com.privateinternetaccess.regions.internals
+import platform.Foundation.NSUserDefaults
 
 internal actual class PersistenceRegionsDataSource(
-    preferenceName: String? = null,
+    preferenceName: String? = "PersistenceRegionsDataSource",
     private val logErrors: Boolean
 ) : RegionsCacheDataSource, AbstractPersistenceRegionsDataSource() {
 
+    private val userDefaults: NSUserDefaults
+
     init {
-        throw UnsupportedOperationException("not yet implemented")
+        userDefaults = NSUserDefaults(suiteName = preferenceName ?: "PersistenceRegionsDataSource")
```

Comment:
nit. We can probably leave the constructor with the default as `null` to avoid the duplication of strings. It achieves the same result:

**Scenario 1.0 (Defaulting to `null`):**

- A preferenceName is not defined. Which means it would be constructed with `null`.
- Once it reaches this line the elvis operator picks it up and uses `PersistenceRegionsDataSource` instead.

**Scenario 1.1 (Defaulting to `null`):**

- A preferenceName is explicitly set to `null`.
- Once it reaches this line the elvis operator picks it up and uses `PersistenceRegionsDataSource` instead.

**Scenario 2.0 (Defaulting to `PersistenceRegionsDataSource`):**

- A preferenceName is not defined. Which means it would be constructed with `PersistenceRegionsDataSource`.
- Once it reaches this line the elvis operator will use `PersistenceRegionsDataSource`.

**Scenario 2.1 (Defaulting to `PersistenceRegionsDataSource`):**

- A preferenceName is explicitly set to `null`.
- Once it reaches this line the elvis operator picks it up and uses `PersistenceRegionsDataSource` instead.

Reply:
Yes, it makes sense.
I also think it's better to avoid the string duplication. I will update it
Thanks